indico/modules/events/registration/util.py

```diff
@@ -384,7 +385,7 @@ def modify_registration(registration, data, management=False, notify_user=True):
 
     billable_items_locked = not management and registration.is_paid
     for form_item in regform.active_fields:
-        if form_item.is_purged:
+        if form_item.is_purged or form_item.is_locked(registration):
```

Comment:
I also didn't call it there because I'm not sure if it would be better to pass `None` or the newly created `Registration` object - I guess it would depend on the potential use case?

Reply:
`None` for sure, especially since you also need to call this when rendering the form to know whether it should be displayed as locked or not - and at that point you do not have such an object.